### PR TITLE
Feat/borrow

### LINKED ISSUE #170: Fix: Chien's skill issue 2

### LINKED ISSUE #172: add: borrow features

### PR DESCRIPTION
**Pull request by ChienNQuang**

### CODE REVIEW

File: src/Application/Borrows/Queries/GetAllBorrowRequestsSpecificPaginated.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                .Paginate(pageNumber.Value, sizeNumber.Value)
+                .OrderByCustom(sortBy, sortOrder)
+                .ToListAsync(cancellationToken);
+            
+            var result = _mapper.Map<List<BorrowDto>>(list);
+
+            return new PaginatedList<BorrowDto>(result, result.Count, pageNumber.Value, sizeNumber.Value);
+        }
+    }
+}
```
Comment (vzart) [inline src/Application/Borrows/Queries/GetAllBorrowRequestsSpecificPaginated.cs:79]: Is this necessary to separate from getAll?

**Review by vzart — COMMENTED** [review]
Nothing seems wrong, wait for kaitoz11 review

File: src/Infrastructure/Services/MailService.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public bool SendResetPasswordHtmlMail(string userEmail, string password)
         request.AddHeader("Content-Type", "application/json");
         request.AddParameter("application/json", json, ParameterType.RequestBody);
         var response = client.Execute(request);
-
+        Console.WriteLine(response.Content);
```
Comment (kaitoz11) [inline src/Infrastructure/Services/MailService.cs]: remove this?

File: src/Application/Borrows/Queries/GetAllBorrowRequestsSpecificPaginated.cs (hunk 1 above)
Comment (kaitoz11) [reply src/Application/Borrows/Queries/GetAllBorrowRequestsSpecificPaginated.cs:79]: same question

File: src/Application/Borrows/Commands/BorrowDocument.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+            RuleFor(x => x.Reason)
+                .MaximumLength(512).WithMessage("Reason cannot exceed 512 characters.");
+
+            RuleFor(x => x.BorrowFrom)
+                .GreaterThan(DateTime.Now).WithMessage("Borrow date cannot be in the past.")
+                .Must((command, borrowTime) => borrowTime < command.BorrowTo).WithMessage("Due date cannot be before borrow date.");
+
+            RuleFor(x => x.BorrowTo)
+                .GreaterThan(DateTime.Now).WithMessage("Due date cannot be in the past.");
```
Comment (kaitoz11) [inline src/Application/Borrows/Commands/BorrowDocument.cs:30]: Is there a way to check BorrowFrom < BorrowTo?

File: src/Application/Borrows/Commands/UpdateBorrow.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+using Domain.Entities.Physical;
+using Domain.Statuses;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace Application.Borrows.Commands;
+
+public class UpdateBorrow
+{
```
Comment (kaitoz11) [inline src/Application/Borrows/Commands/UpdateBorrow.cs:15]: Validator....?

**Review by kaitoz11 — CHANGES_REQUESTED** [review]
nice try

File: src/Application/Borrows/Commands/BorrowDocument.cs (hunk 3 above)
Comment (ChienNQuang) [reply src/Application/Borrows/Commands/BorrowDocument.cs:30]: 2 lines above this line

File: src/Infrastructure/Services/MailService.cs (hunk 2 above)
Comment (ChienNQuang) [reply src/Infrastructure/Services/MailService.cs]: ok

**Review by vzart — APPROVED** [review]
LGTM

File: docker-compose.test.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ services:
       - '8888:80'
     environment:
       - ASPNETCORE_ENVIRONMENT=Testing
-      - PROFILE_DatabaseSettings__ConnectionString=Server=database;Port=5432;Database=mytestdb;User ID=profiletester;Password=supasupasecured;
+      - PROFILE_DatabaseSettings__ConnectionString=Server=database;Port=5432;Database=mytestdb;User ID=profiletester;Password=supasupasecured;Include Error Detail=true
```
Comment (kaitoz11) [inline docker-compose.test.yml:12]: What is this for?

Comment (ChienNQuang) [reply docker-compose.test.yml:12]: the test error message should be clearer

**Review by kaitoz11 — APPROVED** [review]
LGTM

**Pull request merged**